### PR TITLE
WINC-332: Generated file linting

### PR DESCRIPTION
**Pull request by sebsoto**
This PR ensures that all generated files are fully updated as part
of `make lint`.
The bundle files were out of date, so they needed to be updated
in order for linting to pass.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -148,29 +148,45 @@ else
...
 .PHONY: bundle-build ## Build the bundle image.
 bundle-build:
 	podman build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
 
 .PHONY: lint
-lint:
-	hack/lint-gofmt.sh
-	hack/verify-vendor.sh
+lint: osdk
+	#hack/lint-gofmt.sh
```
Comment (jrvaldes) [inline Makefile:187]: Why is this commented out?

File: Makefile [hunk 2]
```diff
@@ -159,6 +159,7 @@ bundle: manifests kustomize
 	operator-sdk generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle --overwrite=false -q --version $(WMCO_VERSION) $(BUNDLE_METADATA_OPTS)
+	sed -i 's/REPLACE_IMAGE:latest/REPLACE_IMAGE/g' bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
```
Comment (aravindhp) [inline Makefile:176]: What does have to do with generated file linting?

File: Makefile (hunk 1 above)
Comment (sebsoto) [reply Makefile]: debugging changes got added by accident :sweat_smile:

File: bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ metadata:
     olm.skipRange: '>=4.0.0 <5.0.0'
     operatorframework.io/cluster-monitoring: "true"
     operatorframework.io/suggested-namespace: openshift-windows-machine-config-operator
-    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
+    operators.operatorframework.io/builder: operator-sdk-v1.14.0+git
```
Comment (aravindhp) [inline bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:14]: Is this change needed for `make lint` to succeed?

File: Makefile [hunk 4]
```diff
@@ -148,28 +148,43 @@ else
...
 endif
 endif
+
+.PHONY: osdk ## Download operator-sdk to /tmp/operator-sdk if it is not present on $PATH
+OSDK = /tmp/operator-sdk
+osdk:
+ifeq (,$(shell which operator-sdk 2>/dev/null))
+	@{ \
+	set -e ;\
+	wget --no-verbose -O $(OSDK) https://github.com/operator-framework/operator-sdk/releases/download/v1.15.0/operator-sdk_linux_amd64 -o operator-sdk;\
```
Comment (aravindhp) [inline Makefile:158]: We continue to download the SDK and oc in CI. We should consider injecting them into the image. Ref: [WINC-520](https://issues.redhat.com/browse/WINC-520).

You could say this is outside the scope of the PR, however you are now introducing yet another spot where we are downloading the SDK.

**Review by aravindhp — COMMENTED** [review]
Thanks for adding this, @sebsoto. PTAL at my comments.

File: bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (sebsoto) [reply bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:14]: It seems like it is.
If I revert the change by adding another commit:
```
[sebsoto@sdesktop windows-machine-config-operator]$ git show HEAD
commit c2a66d4383a37d4075c4e1f96b54ca10f18fc299 (HEAD -> lint_bundle)
Author: Sebastian Soto <ssoto@redhat.com>
Date:   Fri Dec 17 12:36:33 2021 -0500

    f

diff --git a/bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml b/bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
index de7c8b62..cab8700e 100644
--- a/bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
+++ b/bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
@@ -11,7 +11,7 @@ metadata:
     olm.skipRange: '>=4.0.0 <5.0.0'
     operatorframework.io/cluster-monitoring: "true"
     operatorframework.io/suggested-namespace: openshift-windows-machine-config-operator
-    operators.operatorframework.io/builder: operator-sdk-v1.14.0+git
+    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
     operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
     repository: https://github.com/openshift/windows-machine-config-operator
     support: Red Hat
[sebsoto@sdesktop windows-machine-config-operator]$ hack/verify-bundle.sh 
/home/sebsoto/code/openshift/windows-machine-config-operator/bin/controller-gen rbac:roleName=windows-machine-config-operator crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/usr/local/bin/operator-sdk generate kustomize manifests -q
WARN[0000] Skipping definitions parsing: APIs root dir "api" does not exist 
cd config/manager && /home/sebsoto/code/openshift/windows-machine-config-operator/bin/kustomize edit set image controller=REPLACE_IMAGE
/home/sebsoto/code/openshift/windows-machine-config-operator/bin/kustomize build config/manifests | /usr/local/bin/operator-sdk generate bundle --overwrite=false -q --version 5.0.0 --channels="preview,stable" --default-channel="stable"
sed -i 's/REPLACE_IMAGE:latest/REPLACE_IMAGE/g' bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
/usr/local/bin/operator-sdk bundle validate ./bundle
INFO[0000] All validation tests have completed successfully 
sed -i 's/windows-machine-config-operator\.v.\.0\.0/windows-machine-config-operator.v5.0.0/g' ./bundle/windows-machine-config-operator.package.yaml
bundle lint failure
 M bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
```

Comment (aravindhp) [reply bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:14]: My question was to do with the whole commit. Moreover `operator-sdk-v1.14.0+git` seems wrong given the SDK version we are using.

File: Makefile (hunk 2 above)
Comment (sebsoto) [reply Makefile:176]: Without this change, in the verify script make bundle will be adding `:latest` to REPLACE_IMAGE, causing a diff to happen.

We do not want the `:latest` tag to be merged in, but our bundle generation script doesn't reflect that. So the options here were to either
1) change the bundle generation script like I did
2) change the validation script to remove the `:latest` when validating

If I went with 2 it would leave us with the extra work of manually removing the `:latest` tag, so 1 was the clear answer for me.

If you feel this needs to be broken into a separate PR I can do that.

Comment (aravindhp) [reply Makefile:176]: This is the sort of explanation that should be in the commit message 😃

>This is being manually removed when devs are merging bundle changes.

Please remind me why we manually remove this.

Comment (alinaryan) [reply Makefile:176]: > Please remind me why we manually remove this.
We need to remove it to maintain CI compatibility

Comment (aravindhp) [reply Makefile:176]: What do you mean by CI compatibility?

Comment (alinaryan) [reply Makefile:176]: CI doesn't recognize `:latest` after `REPLACE_IMAGE` and causes tests to fail with "error waiting for CSV to install"

Comment (aravindhp) [reply Makefile:176]: That's not CI but the SDK given we are using the SDK to install WMCO. Correct?
